### DISCUSSION TITLE
WrapperGeneration for OpenCascade 7.4.0

### DISCUSSION THREAD

**Discussion by aliveho** [Macad3D/Macad3D#7] [Q&A]
This is a nice work.

Do you have a WrapperGeneration for OpenCascade 7.4.0?
I must use the version currently. 
Or please let me know how to modify your works.

Thank you

**Comment by dipts** [comment]
If you're only interested in the wrapper then you can try to use OCCT 7.4.0 as source for the wrapper generation. Macad itself will need some adjustments in the Macad.Managed module to work with 7.4.0.

Another way is to use older version 72ff890a339435247cfa91057e8df9ed30620cca, which is still based on 7.4.0. Please note that the way OCCT is imported is different, check the readme of that commit.

**Reply by aliveho** [reply]
Thank you for your help. 
I successfully compiled all. But When I run macad.exe, one error happens.

![error](https://user-images.githubusercontent.com/5700640/111029589-d0282f80-8440-11eb-9e33-ed0a1b0b87c8.png)

I expect your kind reply.
Sincerely

**Reply by dipts** [reply]
Did you choosed the way using https://github.com/Macad3D/Macad3D/commit/72ff890a339435247cfa91057e8df9ed30620cca? If yes, you probably hit the missing-dll-issue described [here](https://github.com/Macad3D/Macad3D/discussions/5). Just copy the dlls from OCCT third parties _FreeImage_, _ffmpeg_ and _openvr_ to the application folder.

If you tried to generate wrappers with head and OCCT 7.4.0 - then I don't have a clue, you have to debug whats going on.

**Reply by aliveho** [reply]
I read the discussion for the same problem.

I copied the missing files as a solution in the discussions.
It's OK.
